### ISSUE TITLE
Command Execution Unexpectedly Failed

### ISSUE THREAD

**Issue by MadsMighty** [zekroTJA/dcdl#4]
Hi there,

I feel as though I'm SO CLOSE to getting this bot to do exactly what I'm looking for, but for some reason it won't execute its commands properly.

![Screen Shot 2022-01-06 at 7 24 37 PM](https://user-images.githubusercontent.com/64160144/148486566-9e82ea71-8f10-4420-bddf-67e87d907a34.png)

I have this docker container running alongside hundreds of other functional and reachable-through-reverse-proxy services, and can't find what's preventing the command from running properly. Below is an image of the environment variables I've set up through the Unraid interface, with all volume mounts showing correctly in the file system:

![Screen Shot 2022-01-06 at 7 27 09 PM](https://user-images.githubusercontent.com/64160144/148486758-11aaff32-d5b5-460b-9955-7ef8778f5903.png)

Any help would be really appreciated!

**Comment by zekroTJA** [comment]
I think the problem is that the `Dockerfile` sets `/var/data` as data location for the downloaded archives.

https://github.com/zekroTJA/dcdl/blob/8e57e5487e11aa1b86e3c68fc82931a4c833989e/Dockerfile#L10

Maybe either try to bind your storage location `/mnt/user/appdata/dcdl/data` to `/var/data` or set the `DCDL_STORAGE_LOCATION` env variable to `/root/data` (if I correctly assume that this is the target directory to bind to? Idk, I never worked with Unraid ^^).

**Comment by MadsMighty** [comment]
> I think the problem is that the `Dockerfile` sets `/var/data` as data location for the downloaded archives.
> 
> https://github.com/zekroTJA/dcdl/blob/8e57e5487e11aa1b86e3c68fc82931a4c833989e/Dockerfile#L10
> 
> Maybe either try to bind your storage location `/mnt/user/appdata/dcdl/data` to `/var/data` or set the `DCDL_STORAGE_LOCATION` env variable to `/root/data` (if I correctly assume that this is the target directory to bind to? Idk, I never worked with Unraid ^^).

Once I set my bind mount to /var/data it came online and processed a batch download as expected thank you!

Little quirk though: opening the link DCDL sends you in Firefox just brings a page of random unicode whereas opening the link in Chrome starts downloading the .zip file properly.

Thank you so much again for this project its helping fill a gap on my server that's much appreciated.

**Comment by zekroTJA** [comment]
> Little quirk though: opening the link DCDL sends you in Firefox just brings a page of random unicode whereas opening the link in Chrome starts downloading the .zip file properly.

Oh yeah, that could be because of missing `content-type` headers, most probably. I will fix that as soon as I have the time to do so.

Thank you, I am very happy to hear that you appreciate it. 💖

**Comment by zekroTJA** [comment]
Update:

First of all, I've now added a check routine which runs at startup which checks for the existence of the storage directory. So, you notice on startup that the location is not existing instead of during runtime. Not perfect, but better I guess. ^^

Also, I've added the `Content-Type: application/zip` so that Firefox will correctly download the file when opening the link. 

If you notice any other issues or if you have some ideas or enhancements, feel free let me know. 
Thank you very much for your issue contribution. :)
